### PR TITLE
Automatically remove temporary files in tests

### PR DESCRIPTION
**Pull request by jishnub**
This PR changes various uses of the `try..finally` pattern in tests to `mktemp`, so that temporary files are automatically removed

### CODE REVIEW

**Comment by jishnub** [comment]
Not sure why `fits_clobber_file` fails on windows with file already exists, perhaps some race condition?

File: test/runtests.jl [hunk 1]
```diff
@@ -8,12 +8,8 @@ function tempfitsfile(fn)
         fitsfile = fits_clobber_file(filename)
         fn(fitsfile)
 
-        # Write arbitrary data to file to avoid errors on closing it
         if fitsfile.ptr != C_NULL
-            temp = ones(1)
-            fits_create_img(fitsfile, temp)
-            fits_write_pix(fitsfile, temp)
-            close(fitsfile)
```
Comment (giordano) [inline test/runtests.jl:16]: Are you sure it's safe to remove the call to `close`?  Open handles have always caused problems for me on Windows

File: test/runtests.jl [hunk 2]
```diff
@@ -338,11 +333,6 @@ end
             fits_write_pix(f, a)
             close(f)
 
-            f2 = fits_clobber_file(tempname() * ".fits")
-            fits_create_img(f2, eltype(a), [size(a)...])
-            fits_write_pix(f2, a)
-            close(f2)
```
Comment (giordano) [inline test/runtests.jl:344]: Why was this removed?

File: test/runtests.jl (hunk 1 above)
Comment (jishnub) [reply test/runtests.jl:16]: From what I understand by [reading the docs](https://heasarc.gsfc.nasa.gov/docs/software/fitsio/c/c_user/node35.html), `fits_delete_file` will close and delete the file, ignoring errors if any. This gets around the fact that `fits_close_file` errors while closing an empty file. I'm not totally sure how this works on windows, but tests seem to pass.

File: test/runtests.jl (hunk 2 above)
Comment (jishnub) [reply test/runtests.jl:344]: This was a duplicate occurrence of the same code. The lines https://github.com/JuliaAstro/CFITSIO.jl/blob/e4fe70f7c5b408b0bfd784790cb3ee2c354c6d16/test/runtests.jl#L341-L344 and https://github.com/JuliaAstro/CFITSIO.jl/blob/e4fe70f7c5b408b0bfd784790cb3ee2c354c6d16/test/runtests.jl#L405-L409 essentially did the same thing, so I've cleaned it up.
